src/loops/programs_with_repetition/easy/q5/MainTest.java

```diff
@@ -0,0 +1,95 @@
+package loops.programs_with_repetition.easy.q5;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+
+public class MainTest extends BaseTest {
+  // Java
+  public Clause[] testSentence() {
+    TestOption.isInputTest = true;
+    TestOption.defaultInput = "777";
+    return new Clause[] {
+        new StringLiteral("Enter an integer\\: "),
+        new NewLine(),
+        new PlaceHolder()
+    };
+  }
+
+  public void runMain() {
+    MirrorSeven.main(new String[0]);
+  }
+
+  static Stream<Integer> inputProviderValidOneInput() {
+    return Stream.of(0, 7, 77, 161, 252, 343, 434, 525, 595, 616, 686, 707, 777, 868, 959, 1001, 1771, 2002, 2772, 3003,
```

Comment:
You should not use so many test cases because the more test cases there are, the more points the question is weighted for. I would suggest using about half of these.